htsget-config/README.md

```diff
@@ -14,57 +14,292 @@ Configuration for [htsget-rs] and relevant crates.
+
+Additionally, the resolver component has a feature, which allows resolving IDs based on the other fields present in a query.
+This is useful as allows the resolver to match only match an ID, if a particular set of query parameters are also present. For example, 
+a resolver can be set to only resolve IDs if the format is also BAM.
+
+To set the resolver 'allow guard', add a `[resolver.allow_guard]` table, and set the following options:
+
+| Option                  | Description                                                                             | Type                                                                  | Default                             |
+|-------------------------|-----------------------------------------------------------------------------------------|-----------------------------------------------------------------------|-------------------------------------|
+| `allow_reference_names` | Resolve the query ID if the query also contains the reference names set by this option. | Array of reference names or `'All'`                                   | `'All'`                             | 
+| `allow_fields`          | Resolve the query ID if the query also contains the fields set by this option.          | Array of fields or `'All'`                                            | `'All'`                             |
+| `allow_tags`            | Resolve the query ID if the query also contains the tags set by this option.            | Array of tags or `'All'`                                              | `'All'`                             |
+| `allow_formats`         | Resolve the query ID if the query is one of the formats specified by this option.       | An array of formats containing `'BAM'`, `'CRAM'`, `'VCF'`, or `'BCF'` | `['BAM', 'CRAM', 'VCF', 'BCF']`     |
+| `allow_classes`         | Resolve the query ID if the query is one of the classes specified by this option.       | An array of classes containing eithr `'body'` or `'header'`           | `['body', 'header']`                |
+| `allow_interval_start`  | Resolve the query ID if the query reference start position is at least this option.     | Unsigned 32-bit integer start position, 0-based, inclusive            | Not set, allows all start positions |
+| `allow_interval_end`    | Resolve the query ID if the query reference end position is at most this option.        | Unsigned 32-bit integer end position, 0-based exclusive.              | Not set, allows all end positions   |
+
+An example of a fully configured resolver:
+
+```toml
+[[resolvers]]
+regex = '.*'
+substitution_string = '$0'
+
+[resolvers.storage_type]
+type = 'S3'
+bucket = 'bucket'
+
+[resolvers.allow_guard]
+allow_reference_names = ['chr1']
+allow_fields = ['QNAME']
+allow_tags = ['RG']
+allow_formats = ['BAM']
+allow_classes = ['body']
+allow_interval_start = 100
+allow_interval_end = 1000
+```
+
```

Comment:
So for example, this resolver would... only allow queries to chr1 between positions 100 and 1000 on BAMs?

Reply:
Yes, I'll clarify that a bit in the docs.